src/routes/admin/sources.ts

```diff
@@ -117,32 +119,32 @@ export function buildAdminSourcesRoutes(deps) {
 
     async function createSource(request, reply) {
         const body = request.body;
-        let errors = null;
+        let errors: ErrorResponseList = null;
         const missingParams = [
             'name',
         ].filter(param => !body[param]);
 
         if (missingParams.length > 0) {
-            errors = encodeError(errors, ERROR_PARAMS_MISSING.code, ERROR_PARAMS_MISSING.message, {params: missingParams});
+            errors = encodeErrorFromType(errors, ERROR_PARAMS_MISSING.type, {params: missingParams});
         }
 
         if (errors) {
             return void reply.code(400).send(errors);
         }
 
         try {
-            const invalidParams = [];
+            const invalidParams: object[] = [];
             const flows = body.flows;
             if (flows) {
                 if (Array.isArray(flows)) {
                     // Check each flow
                     for (const flow of flows) {
                         const targetId = flow.targetId;
-                        const target = await targetsService.getTargetById(targetId);
+                        const target = await targetsService!.getTargetById(targetId);
```

Comment:
I am not sure why you need to always use the exclamation mark any time you need to get access to service functions.